crossbeam-channel/src/channel.rs

```diff
@@ -393,6 +394,43 @@ impl<T> Sender<T> {
         }
     }
 
+    /// Attempts to send a message into the channel without blocking and without performing any work
+    /// if sending fails. This function uses a [Cow](https://doc.rust-lang.org/std/borrow/enum.Cow.html)
+    /// smart pointer to avoid the need for cloning the message contents if the receiving side is full
+    /// or disconnected.
+    ///
+    /// This method will either send a message into the channel immediately or return an error if
+    /// the channel is full or disconnected. The returned error contains the original Cow.
+    ///
+    /// If called on a zero-capacity channel, this method will send the message only if there
+    /// happens to be a receive operation on the other side of the channel at the same time.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::borrow::Cow;
+    /// use crossbeam_channel::{bounded, TrySendError};
+    ///
+    /// let (s, r) = bounded(1);
+    ///
+    /// assert_eq!(s.try_send_cow(Cow::Borrowed(&1)), Ok(())); // Clone occurs
+    /// assert_eq!(s.try_send_cow(Cow::Owned(1)), Ok(())); // Clone does not occur
+    /// assert_eq!(s.try_send_cow(Cow::Borrowed(&2)), Err(TrySendError::Full(Cow::Borrowed(&2)))); // Clone does not occur
+    ///
+    /// drop(r);
+    /// assert_eq!(s.try_send_cow(Cow::Borrowed(&3)), Err(TrySendError::Disconnected(Cow::Borrowed(&3)))); // Clone does not occur
+    /// ```
+    pub fn try_send_cow<'a>(&self, msg: Cow<'a, T>) -> Result<(), TrySendError<Cow<'a, T>>>
+    where
+        T: Clone
```

Comment:
This trait bound is redundant btw.

Reply:
Is it?

![image](https://user-images.githubusercontent.com/14863743/125208343-9bf73b00-e289-11eb-97ec-5d12d466ca9f.png)

Reply:
`Clone` should probably be changed to `ToOwned`, really.

Reply:
Oh yes ofc. Sorry you are right :)

Can we perhaps do something like this?:

```rust

impl<T: Clone> Sender<T> {
    // Put all clone related methods here
    pub fn try_send_cow<'a>(&self, msg: Cow<'a, T>) -> Result<(), TrySendError<Cow<'a, T>>> {...}
}

```

so we don't have to keep adding that `Clone` bound on every method that needs it.

Reply:
Absolutely, that's a good idea.